relay-general/src/store/normalize.rs

```diff
@@ -344,166 +211,355 @@ impl<'a> NormalizeProcessor<'a> {
+                    *user_ip = client_ip.to_owned();
+                }
+            }
+        }
     }
 
-    fn normalize_user_agent(&self, _event: &mut Event) {
-        if self.config.normalize_user_agent.unwrap_or(false) {
-            #[cfg(feature = "uaparser")]
-            user_agent::normalize_user_agent(_event);
-
-            #[cfg(not(feature = "uaparser"))]
-            panic!("relay not built with uaparser feature");
+    // Copy IPs from request interface to user, and resolve platform-specific backfilling
+    let http_ip = event
+        .request
+        .value()
+        .and_then(|request| request.env.value())
+        .and_then(|env| env.get("REMOTE_ADDR"))
+        .and_then(Annotated::<Value>::as_str)
+        .and_then(|ip| IpAddr::parse(ip).ok());
+
+    if let Some(http_ip) = http_ip {
+        let user = event.user.value_mut().get_or_insert_with(User::default);
+        user.ip_address.value_mut().get_or_insert(http_ip);
+    } else if let Some(client_ip) = client_ip {
+        let user = event.user.value_mut().get_or_insert_with(User::default);
+        // auto is already handled above
+        if user.ip_address.value().is_none() {
+            let platform = event.platform.as_str();
+
+            // In an ideal world all SDKs would set {{auto}} explicitly.
+            if let Some("javascript") | Some("cocoa") | Some("objc") = platform {
+                user.ip_address = Annotated::new(client_ip.to_owned());
+            }
         }
     }
 }
 
+pub fn light_normalize_event(
+    event: &mut Annotated<Event>,
+    client_ip: Option<&IpAddr>,
+    user_agent: Option<&str>,
+    received_at: Option<DateTime<Utc>>,
+    max_secs_in_past: Option<i64>,
+    max_secs_in_future: Option<i64>,
+    breakdowns_config: Option<BreakdownsConfig>,
+) -> ProcessingResult {
+    event.apply(|event, meta| {
+        // Process security reports first to ensure all props.
+        normalize_security_report(event, client_ip, user_agent);
+
+        // Insert IP addrs before recursing, since geo lookup depends on it.
+        normalize_ip_addresses(event, client_ip);
+
+        // Validate the basic attributes we extract metrics from
+        event.release.apply(|release, meta| {
+            if protocol::validate_release(release).is_ok() {
+                Ok(())
+            } else {
+                meta.add_error(ErrorKind::InvalidData);
+                Err(ProcessingAction::DeleteValueSoft)
+            }
+        })?;
+        event.environment.apply(|environment, meta| {
+            if protocol::validate_environment(environment).is_ok() {
+                Ok(())
+            } else {
+                meta.add_error(ErrorKind::InvalidData);
+                Err(ProcessingAction::DeleteValueSoft)
+            }
+        })?;
+
+        // Default required attributes, even if they have errors
+        normalize_release_dist(event); // dist is a tag extracted along with other metrics from transactions
```

Comment:
Do we need all these comments here?

Reply:
I asked for these comments because I figured it might be helpful to know in the future why we decided to pull those into "light" normalization.